script2.js

```diff
@@ -91,204 +71,247 @@ function Board (options){
 
-  function controlComputerMove (){
-    let cell = getComputerMove()
-    let symbol = (model.game.isPlayer1Next) ? model.symbols.player1 : model.symbols.player2;
-    applyMove(cell.row, cell.column, symbol)
-    model.game.result = getResult(model.game._gameBoard)
+  let resultText = "tie"
+  if(result !== RESULT.tie)
+    resultText = getPlayerName(result) + " won"
 
-    if (model.game.result.result ==RESULT.incomplete){
-      setNextMove()
-      render(VIEW.game)
-    }
-    else {
-      if (model.game.result.result==RESULT.player2Won) {model.totalscore.player2++}
-      render(VIEW.result)
-    }
+
+  let htmlBefore = `<p>${resultText} &emsp; &ensp; Click to restart game </p> `
+  let board = model.game._gameBoard.reduce(function(acc,curr,rowIndex){
+      return acc + `<div class="row">${curr.map((str,itemIndex)=>`<div class="cell" data-row=${rowIndex} data-column=${itemIndex}>${str}</div>`).join('')}</div>`
+    }, ``)
+  let htmlAfter = `<p>Score: Player 1 - ${model.players[0].score} &emsp; &ensp; Player 2 - ${model.players[1].score}</p>`
+  return `<div id='resultView'> ${htmlBefore} <div id="board">${board}</id> ${htmlAfter} </div>`
+}
+
+function render(view){
+  // console.log('renderS')
+  let html = ''
+  if (view == VIEW.question1) {html = htmlQ1()}
+  else if (view == VIEW.question2) {html = htmlQ2()}
+  else if (view == VIEW.result) {html=htmlGameEnd()}
+  else {html=htmlGame()}
+  // console.log(html)
+  options.el.innerHTML = html
+}
+
+
+function question1Handler (ev){
+  model.players[1].isComputer = !($(ev.currentTarget).attr('data')==="2players")
+  render(VIEW.question2)
+}
+
+function question2Handler (ev){
+  let player1Symbol = $(ev.currentTarget).attr('data')
+  model.players[0].symbol=player1Symbol;
+  model.players[1].symbol=(player1Symbol===SYMBOLS.x)? SYMBOLS.o: SYMBOLS.x;
+
+  if(model.players[model.game.turn].isComputer)
+    doComputerMove()
+  
+  beginGame()
+}
+
+function doComputerMove (){
+  let move = getComputerMove(model.game._gameBoard)
+  executeTurn(model.game._gameBoard, move)
+}
+
+
+function playerMoveHandler (ev){
+  let symbol = model.players[model.game.turn].symbol
+  let row = parseInt($(ev.currentTarget).attr('data-row'))
+  let column = parseInt($(ev.currentTarget).attr('data-column')) 
+  executeTurn(model.game._gameBoard, {row, column, symbol})
+}
+
+function applyMove(board,move) {
+  if (board[move.row][move.column]!==""){
+    return board
   }
 
+  board[move.row][move.column]= move.symbol
+  return board
+}
 
-  function controlGame (ev){
-    let symbol = (model.game.isPlayer1Next) ? model.symbols.player1 : model.symbols.player2;
-    let row = $(ev.currentTarget).attr('data-row')
-    let column = $(ev.currentTarget).attr('data-column')
-    applyMove(model.game._gameBoard, row, column, symbol)
-    model.game.result = getResult(model.game._gameBoard)
+function executeTurn(board, move) {
+  // applyMove(model.game._gameBoard, row, column, symbol)
+  applyMove(board,move)
```

Comment:
Isn't it strange that you return a board, but then you don't "catch" it? 
If u change the board directly since it is just a reference, can u just return nothing?